source/jormungandr/jormungandr/routing/valhalla.py

```diff
@@ -0,0 +1,190 @@
+#  Copyright (c) 2001-2014, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from __future__ import absolute_import, print_function, unicode_literals, division
+from navitiacommon import request_pb2, response_pb2, type_pb2
+import logging
+from jormungandr.routing.abstract_routing import AbstractRouting
+import pybreaker
+import requests as requests
+from jormungandr import app
+import json
+from flask_restful import abort
+from exceptions import NotImplementedError
+from jormungandr.exceptions import UnableToParse, TechnicalError
+
+
+class Valhalla(AbstractRouting):
+
+    def __init__(self, instance, **kwargs):
+        self.instance = instance
+        self.args = kwargs
+        self.breaker = pybreaker.CircuitBreaker(fail_max=app.config['CIRCUIT_BREAKER_MAX_VALHALLA_FAIL'],
+                                                reset_timeout=app.config['CIRCUIT_BREAKER_VALHALLA_TIMEOUT_S'])
+        self.convert_distances = {'kilometers': 1000.0, 'miles': 1609.34}
```

Comment:
You can change the request unit used by valhalla: https://mapzen.com/documentation/mobility/turn-by-turn/api-reference/#directions-options